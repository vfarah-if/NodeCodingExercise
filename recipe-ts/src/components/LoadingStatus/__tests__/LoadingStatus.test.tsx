import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

import {
	DefaultLoadingStatus,
	InvisibleLoadingStatus,
} from '../LoadingStatus.stories';

describe('LoadingStatus', () => {
	test('should load default status when no value assigned', () => {
		const { container } = render(<DefaultLoadingStatus isLoading={true} />);

		expect(container).toMatchSnapshot();
	});

	test('should be invisible when not loading', () => {
		const { container } = render(
			<InvisibleLoadingStatus isLoading={false} />
		);

		expect(container).toMatchSnapshot();
	});
});
