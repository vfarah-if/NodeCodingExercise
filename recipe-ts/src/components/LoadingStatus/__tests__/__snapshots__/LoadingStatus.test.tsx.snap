// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingStatus should be invisible when not loading 1`] = `<div />`;

exports[`LoadingStatus should load default status when no value assigned 1`] = `
<div>
  <div
    class="loader"
  />
</div>
`;
